src/mbgl/renderer/buckets/circle_bucket.hpp

```diff
@@ -16,6 +16,7 @@ class BucketParameters;
 class CircleBucket : public Bucket {
 public:
     CircleBucket(const BucketParameters&, const std::vector<const RenderLayer*>&);
+    ~CircleBucket() override;
```

Comment:
nit: declare as `final`? Also, for the rest of buckets.

Reply:
:+1: however, making the whole class `final` makes it more readable